src/main/com/fulcrologic/rad/routing/html5_history.cljc

```diff
@@ -86,12 +86,14 @@
    parameter is provided the mode is autodetected from presence of hash segment in URL.
   "
   ([] (url->route #?(:clj  false
-                     :cljs (some? (seq (.. js/document -location -hash))))))
-  ([hash-based?]
+                     :cljs (some? (seq (.. js/document -location -hash)))) nil))
+  ([hash-based? prefix]
    #?(:cljs
       (let [path   (if hash-based?
                      (str/replace (.. js/document -location -hash) #"^[#]" "")
                      (.. js/document -location -pathname))
+            pcnt   (count prefix)
+            path   (subs path pcnt)
```

Comment:
`subs` is a bit dangerous when unguarded.  If a user supplies a URL that doesn't include your prefix and it makes it through the stack, then this can throw a `StringIndexOutOfBoundsException`. A safer alternative would be to use `(str/replace path prefix "")` and just get rid of the prefix that way. Then errant input can't cause an exception at least.

Reply:
I would rather stay with subs, str/replace replaces **all**  occurrences 
(str/replace "/aB/aB" "/a" "") => "BB". 
I am going to guard it by starts-with.